Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -1139,9 +1139,9 @@ def _exec_multitest(self,
                        "-logfile", f"@log@/{log_name}",
                        "-project-log-path", log_path_function(run_id, workspace)] + test_cmdline_args
         # MaterialEditor
-        elif type(executable) in [WinMaterialEditor, LinuxMaterialEditor]:
-            log_path_function = editor_utils.retrieve_material_editor_log_path
-            log_content_function = editor_utils.retrieve_material_editor_log_content
+        elif type(executable) in [WinMaterialEditor, LinuxMaterialEditor, WinMaterialCanvas, LinuxMaterialCanvas]:
```

Comment:
Seems like it would be better to compose a parent-type for AtomTools executables, instead of needing to list them all out here as they get implemented. Sounds like the only notable differences are the executable path and where their log is. Then there shouldn't be as much duplication between what's implemented in LyTT for MaterialEditor and MaterialCanvas.

Reply:
+1 I think this is what my other questions were alluding to, some way to consolidate the duplication. But already a great start.